tiny-firmware/firmware/fsm_bitcoin_impl.c

```diff
@@ -50,26 +51,49 @@
 #include "tiny-firmware/firmware/skyparams.h"
 #include "fsm_bitcoin_impl.h"
 
-ErrCode_t msgBitcoinAddressImpl(BitcoinAddress *msg, ResponseSkycoinAddress *resp) {
-    uint8_t seckey[32] = {0};
-    uint8_t pubkey[33] = {0};
-    uint32_t start_index = !msg->has_start_index ? 0 : msg->start_index;
-    if (!protectPin(true)) {
-        return ErrPinRequired;
-    }
-    if (msg->address_n > 99) {
-        return ErrTooManyAddresses;
-    }
+#define FROMHEX_MAXLEN 512
 
-    if (storage_hasMnemonic() == false) {
-        return ErrMnemonicRequired;
+const uint8_t* fromhex(const char* str)
```

Comment:
Move this function to skycoin_crypto.h, as we have tohex there

Reply:
Also, then you can remove fromhex function from skycoin-api/test_skycoin_crypto.c